protocol.md

```diff
@@ -162,6 +163,67 @@ Receive Firmware Name and Version (after query)
 N  END_SYSEX         (0xF7)
 ```
 
+Report supported Firmata features
+---
+
+Report which optional Firmata features are supported by the firmware and the version of each feature. 
+
+As described in the [Feature registry documentation](https://github.com/firmata/protocol/blob/master/feature-registry.md),
+Firmata features are split into 2 categories: *Core features* and *Optional features*. The feature
+query reports the optional features that are supported by the firmware as well as the version
+of each supported optional feature.
+
+The feature query reduces the need to define a new pin mode for every introduced feature. Going
+forward pin modes will only be defined and maintained for microcontroller specific features such as 
+digital and analog I/O, PWM, I2C, UART, SPI, DAC, etc. The use of pin modes for non microcontroller
```

Comment:
> To accomplish this, we would need to associate a unique identifier with a particular board

I worry about trying to maintain something like this, especially when there are likely already several Firmata compatible boards I'm not even familiar with (because compatibility is mostly due to a board's MCU architecture rather than board schematic). I think thoroughly describing the features of the HW would adequately cover this use case and a generic schematic could even be automatically generated if that would be helpful to the user.